sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Controller, Get, Param, ParseIntPipe, Query } from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, HasLocationAccess } from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import { getUserFullName } from "../../utilities";
+import {
+  OfferingChangePaginationOptionsAPIInDTO,
+  PaginatedResultsAPIOutDTO,
+} from "../models/pagination.dto";
+import BaseController from "../BaseController";
+import { ApplicationOfferingChangeSummaryAPIOutDTO } from "./models/application-offering-change-request.institutions.dto";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+
+/**
+ * Application offering change request controller for institutions client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("location/:locationId/application-offering-change-request")
+@ApiTags(
+  `${ClientTypeBaseRoute.Institution}-location-application-offering-change-request`,
+)
+@HasLocationAccess("locationId")
+export class ApplicationOfferingChangeRequestInstitutionsController extends BaseController {
+  constructor(
+    private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService,
+  ) {
+    super();
+  }
+
+  /**
+   * Gets all eligible applications that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible applications that can be requested for
+   * application offering change.
+   */
+  @Get("available")
+  async getEligibleApplicationOfferingChangeRecords(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Query() pagination: OfferingChangePaginationOptionsAPIInDTO,
+  ): Promise<
+    PaginatedResultsAPIOutDTO<ApplicationOfferingChangeSummaryAPIOutDTO>
+  > {
+    const applications =
+      await this.applicationOfferingChangeRequestService.getEligibleApplicationOfferingChangeRecords(
+        locationId,
+        pagination,
+      );
+
+    return {
+      results: applications.results.map((eachApplication) => {
+        const offering = eachApplication.currentAssessment?.offering;
+        return {
+          applicationNumber: eachApplication.applicationNumber,
+          applicationId: eachApplication.id,
+          studyStartPeriod: offering?.studyStartDate,
```

Comment:
`studyStartDate` and `studyEndDate` cannot be null anymore (even DB states otherwise) and they definitely cannot be null for completed applications.
I would prefer that these ones and the DTO would express them as required.

Reply:
we do was some old data with no offerings, which is causing below error, 
![image](https://github.com/bcgov/SIMS/assets/77353155/eab12188-5ecc-4747-a6c5-dd49bc5cdfdb)
so, adding the `?` back @andrewsignori-aot